src/@next/pages/CheckoutPage/utils.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { NextRouter } from "next/router";
+import { ParsedUrlQueryInput } from "querystring";
+
+import type { CartSummaryProps } from "@components/organisms";
+import type { IItems } from "@saleor/sdk/lib/api/Cart/types";
+import { checkoutMessages } from "@temp/intl";
+import {
+  CheckoutStep,
+  CheckoutStepDefinition,
+  CHECKOUT_STEPS,
+} from "@temp/core/config";
+import { checkIfShippingRequiredForProducts } from "@utils/core";
+import { paths } from "@paths";
+
+export type SubpageCompleteHandler = () => void | Promise<void>;
+
+export interface SubpageBaseProps {
+  changeSubmitProgress: (submitInProgress: boolean) => void;
+  onSubmitSuccess: (pageStep: CheckoutStep, data?: object) => void;
+}
+
+export const prepareCartSummaryProducts = (
+  items?: IItems
+): CartSummaryProps["products"] | undefined =>
+  items?.map(({ id, variant, totalPrice, quantity }) => ({
+    id: id || "",
+    name: variant.product?.name || "",
+    price: {
+      gross: {
+        amount: totalPrice?.gross.amount || 0,
+        currency: totalPrice?.gross.currency || "",
+      },
+      net: {
+        amount: totalPrice?.net.amount || 0,
+        currency: totalPrice?.net.currency || "",
+      },
+    },
+    quantity,
+    sku: variant.sku || "",
+    thumbnail: {
+      alt: variant.product?.thumbnail?.alt || undefined,
+      url: variant.product?.thumbnail?.url,
+      url2x: variant.product?.thumbnail2x?.url,
+    },
+  }));
+
+export const getContinueButtonText = (
+  step: CheckoutStep
+): JSX.Element | undefined =>
+  ({
+    [CheckoutStep.Address]: (
+      <FormattedMessage {...checkoutMessages.addressNextActionName} />
+    ),
+    [CheckoutStep.Shipping]: (
+      <FormattedMessage {...checkoutMessages.shippingNextActionName} />
+    ),
+    [CheckoutStep.Payment]: (
+      <FormattedMessage {...checkoutMessages.paymentNextActionName} />
+    ),
+    [CheckoutStep.Review]: (
+      <FormattedMessage {...checkoutMessages.reviewNextActionName} />
+    ),
+  }[step]);
```

Comment:
This object doesn't need to be created on every function call, I'd suggest moving it just above the fn itself:

```suggestion

const continueButtonTextMap: Partial<Record<CheckoutStep, JSX.Element>> = {
  [CheckoutStep.Address]: (
    <FormattedMessage {...checkoutMessages.addressNextActionName} />
  ),
  [CheckoutStep.Shipping]: (
    <FormattedMessage {...checkoutMessages.shippingNextActionName} />
  ),
  [CheckoutStep.Payment]: (
    <FormattedMessage {...checkoutMessages.paymentNextActionName} />
  ),
  [CheckoutStep.Review]: (
    <FormattedMessage {...checkoutMessages.reviewNextActionName} />
  ),
};
export const getContinueButtonText = (step: CheckoutStep) =>
  continueButtonTextMap[step];
```